server/core.ts

```
@@ -111,30 +337,137 @@ export class WebSocketCore {
+
       case 'update':
         if (!id) {
           this.sendError(ws, 'Missing player ID');
           return;
         }
+        // Direct assignment - no conversion needed since we use plain objects
+        
         this.updatePlayer(id as string, restData);
+        // Include laser data if provided
+        const updateData: any = { id, ...restData };
+        if (restData.lasers !== undefined) {
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)